grammars/php.cson

```diff
@@ -2841,6 +2853,68 @@
         'name': 'punctuation.separator.delimiter.php'
       }
     ]
+  'php_doc_types_array_shape':
+    # array shape: {key1: string, key2: int}
+    'begin': '\\{'
+    'beginCaptures':
+      '0':
+        'name': 'punctuation.definition.type.begin.bracket.curly.phpdoc.php'
+    'end': '(\\})'
+    'endCaptures':
+      '1':
+        'name': 'punctuation.definition.type.end.bracket.curly.phpdoc.php'
+    'patterns': [
+      {
+        'include': '#php_doc_types_array_multiple'
+      }
+      {
+        'include': '#php_doc_types_array_single'
+      }
+      {
+        "include": "#php_doc_types_array_with_key_type"
+      }
+      {
+        "include": "#php_doc_types_array_shape"
+      }
+      {
+        'include': '#php_doc_types'
+      }
+      {
+        'match': '[,]\\s*'
+        'name': 'punctuation.separator.delimiter.php'
+      }
+    ]
+  'php_doc_types_array_with_key_type':
+    # array with type key: array<string, int>
+    'begin': '\\<'
```

Comment:
I think I'd like to be a bit more strict with opening brackets, since they could be used in string literals, or add also parsing of string literals.

```php
/**
 * @param ("<"|"{") $startCharacter
 */
```

Reply:
Great point. I added a requirement that it has `array` in front. How does that look?